src/lib.rs

```diff
@@ -5,6 +5,7 @@
 #![allow(clippy::redundant_closure)]
 #![allow(clippy::type_complexity)]
 #![feature(async_await)]
+#![feature(async_closure)]
```

Comment:
It won't be stablized in the near future, will it? Let's avoid it first.

Reply:
It is kind of hard to avoid :-( I think lets wait until async/await is stable and then remove it if async_closure is unlikely to stabilise soon.

Reply:
I worry about this change as well. :(